front/lib/client/handle_file_upload.ts

```diff
@@ -1,19 +1,40 @@
-import type { Result } from "@dust-tt/types";
+import type { Result, SupportedContentFragmentType } from "@dust-tt/types";
+import {
+  isSupportedTextContentFragmentType,
+  supportedTextContentFragment,
+} from "@dust-tt/types";
 import { Err, Ok } from "@dust-tt/types";
 // @ts-expect-error: type package doesn't load properly because of how we are loading pdfjs
 import * as PDFJS from "pdfjs-dist/build/pdf";
-PDFJS.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${PDFJS.version}/pdf.worker.mjs`;
 
-const supportedFileExtensions = [".txt", ".pdf", ".md", ".csv", ".tsv"];
+import { getMimeTypeFromFile } from "@app/lib/file";
+PDFJS.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/${PDFJS.version}/pdf.worker.mjs`;
 
-export async function handleFileUploadToText(
-  file: File
-): Promise<Result<{ title: string; content: string }, Error>> {
+export async function handleFileUploadToText(file: File): Promise<
+  Result<
+    {
+      title: string;
```

Comment:
Any reason for not using `UploadedContentFragment` type?

Reply:
Initially we were not returning the `File` object. I sticked to this logic